### ISSUE TITLE
Lane Arrow tool not notifying user of issues

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#1239]
In code below, it looks like **Lane Arrow** tool should warn me if I try altering a junction which has **lane connectors**:

https://github.com/CitiesSkylinesMods/TMPE/blob/f9798c06fd867d36c174add718ccbefe000ac2c5/TLM/TLM/UI/SubTools/LaneArrows/LaneArrowTool.cs#L295-L310

However, I don't see any on-screen update when attempting to perform such actions.

**Enable turorial messages** is enabled in General settings.

Steps to reproduce:

1. Create a crossroads using vanilla road assets
2. Set up any lane connectors, eg:
    ![image](https://user-images.githubusercontent.com/1386719/146687026-b0c82163-2c72-482d-8896-f51b23748c48.png)
3. Switch to lane arrow tool:
    * I can still select segments at the junction
    * I can still click buttons
    * I neither case is any warning shown about presence of lane connectors

**Comment by kianzarrin** [comment]
it works if you turn on guide. after we wrote this code CO made a change that either you will have all guides+tutorials every time you start a new city or you will have no guide at all. ppl will go with the latter option rendering the guide system useless.

This also influences TMPE tutorials. we should have a system to override the guide system or just abandon it and write our own framework.

**Comment by kianzarrin** [comment]
see it still works :
![image](https://user-images.githubusercontent.com/26344691/146689047-f94daadc-8616-4811-8c1c-2361fc4e24df.png)

**Comment by aubergine10** [comment]
Ah, so both guide options need enabling... ugh.

I'll create specific issue for edcision making on that.

**Comment by aubergine10** [comment]
As for the ad-hoc popups specific to Lane Arrows tool, maybe we could highlight lane red and set mouse (or arrow button?) tooltip to explain what went wrong?

Better would be set arrow button icon to lane connector icon or highway icon (re-use vanilla roads panel highways tab icon?) and give the button a tooltip that explains the problem?

**Comment by aubergine10** [comment]
Regrding th highway rules issue, can highway rules be implemented same way as dedicated turning lanes (and also refresh paths like we do with dedicated turning lanes)? That way we wouldn't need to worry about those and they could be overriden using lane arrows tool too.

**Comment by kianzarrin** [comment]
> can highway rules be implemented same way as dedicated turning lanes

highway merging to outer-most lane only would be an issue. also I cannot tell roundabout from highway entry each of them should act differently (roundabout mass edit does put lane changing rule to overcome this problem so maybe that is enough?)

BTW when you are merging from alley to a high priority road, you should also go to the outer-most lane unless if there is no up-coming traffic.

> and also refresh paths like we do with dedicated turning lanes

with dedicated turning lanes I don't refresh paths. only routings. existing cars will ignore new lane arrows but newly spawned cars will follow the rules. recalculating all paths might take a long time.

**Comment by aubergine10** [comment]
> BTW when you are merging from alley to a high priority road, you should also go to the outer-most lane unless if there is no up-coming traffic.

Yup, high priority road is basically a slow version of a 2-way highway.

> with dedicated turning lanes I don't refresh paths. only routings. existing cars will ignore new lane arrows but newly spawned cars will follow the rules. recalculating all paths might take a long time.

Thanks for clarification, I'll make a note of that in the docs.
